interpreter/valid/valid.ml

```diff
@@ -125,6 +125,10 @@ let check_limits {min; max} range at msg =
     require (I64.le_u min max) at
       "size minimum must not be greater than maximum"
 
+let check_def_type (c : context) (t : def_type) at =
+  match t with
+  | DefT (RecT sts, i) -> assert Int32.(compare i (of_int (List.length sts)) < 0)
```

Comment:
```suggestion
  | DefT (RecT sts, i) -> assert (i < Lib.List32.length sts)
```
But this isn't actually checking anything (only asserting), it seems to be redundant, and you could just do nothing at the call site?

Reply:
I did not know about `Lib.List32`, that is useful. I have fixed this.

I `assert`ed the condition, since I am pretty sure an error at this location is an internal error and not an error the end-user can fix. I tried to think of an error message to provide to the user, but I did not find a good error message. What would you propose? (I could also remove the `assert` and do nothing; I would still prefer to keep the function call, much like eg. `check_num_type`).